assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * AdSense Setup UseSnippet Switch Component.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useEffectOnce } from 'react-use';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { MODULES_ADSENSE } from '../../../datastore/constants';
+import { parseAccountID } from '../../../util/parsing';
+import { UseSnippetSwitch } from '../../common';
+const { useSelect, useDispatch } = Data;
+
+export default function SetupUseSnippetSwitch() {
+	const originalUseSnippet = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getOriginalUseSnippet()
+	);
+	const existingTag = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getExistingTag()
+	);
+	const clientID = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getClientID()
+	);
+
+	const { setUseSnippet } = useDispatch( MODULES_ADSENSE );
+
+	useEffectOnce( () => {
+		if ( existingTag ) {
+			setUseSnippet( false );
+		}
+	} );
```

Comment:
I have some concerns that this hook will work correctly in "production" because the first call to the `getExistingTag` selector will always return `undefined` since the existing tag information is not "resolved" yet. It means that this hook will not run again once the existing tag information is collected and `existingTag` changes to either a string or `null` because `useEffectOnce` is an equivalent of the `useMount` hook.

What we need to do here is to ensure that we execute this hook when the existing tag is resolved.